agent/consul/state/txn.go

```diff
@@ -217,11 +217,14 @@ func (s *Store) txnService(tx *memdb.Txn, idx uint64, op *structs.TxnServiceOp)
 	case api.ServiceGet:
 		entry, err = s.getNodeServiceTxn(tx, op.Node, op.Service.ID, &op.Service.EnterpriseMeta)
 		if entry == nil && err == nil {
-			err = fmt.Errorf("service %q on node %q doesn't exist", op.Service.ID, op.Node)
+			return nil, fmt.Errorf("service %q on node %q doesn't exist", op.Service.ID, op.Node)
 		}
 
 	case api.ServiceSet:
 		err = s.ensureServiceTxn(tx, idx, op.Node, &op.Service)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This `switch` statement is now in a strange place imho, because some errors are returned from the case statement and others not. 

After this `switch` statement, `err` is checked and returned, doesn't that also account for this error?

Reply:
> After this switch statement, err is checked and returned, doesn't that also account for this error?

It does not, because `err` is set to a new value on the line below this.

I will push another commit which moves all the error handling into the switch, removing the need for the trailing `if err != nil` check. Handling errors outside of a block like that is somewhat error prone.

Reply:
🤦 Well, I could've spotted that myself.

Reply:
👍 re the error checks in each block.

Reply:
I pushed another commit with that change. While working on it I realized there is another case where the error from `ensureServiceTxn` is ignored. In the `ServiceCAS` block the err value is overridden, but the linter doesn't notice because there is an `err == nil` check.

I will address that in a follow up PR because I suspect the correct fix for this one is to change the return value of `ensureServiceCASTxn`, which is a slightly larger change.